libraries/AP_EFI/AP_EFI_Serial_Hirth.cpp

```diff
@@ -0,0 +1,328 @@
+}
+
+
+/**
+ * @brief sends the new throttle command to Hirth ECU
+ * 
+ * @param thr - new throttle value given by SRV_Channel::k_throttle
+ * @return true - if success
+ * @return false - currently not implemented
+ */
+bool AP_EFI_Serial_Hirth::send_target_values(uint16_t thr) {
+    int idx = 0;
+    uint8_t computed_checksum = 0;
+
+    //clear buffer
+    for (size_t i = 0; i < sizeof(raw_data); i++)
+    {
+        raw_data[i] = 0;
+    }
+    
+    // Get throttle value
+    uint16_t throttle = thr * THROTTLE_POSITION_FACTOR;
+
+    // set Quantity + Code + "20 bytes of records to set" + Checksum
+    computed_checksum += raw_data[idx++] = req_data.quantity = QUANTITY_SET_VALUE;
+    computed_checksum += raw_data[idx++] = req_data.code = CODE_SET_VALUE;
+    computed_checksum += raw_data[idx++] = throttle & 0xFF;
+    computed_checksum += raw_data[idx++] = (throttle >> 0x08) & 0xFF;
+
+    for (; idx < QUANTITY_SET_VALUE - 2; idx++) {
+        // 0 has no impact on checksum
+        raw_data[idx] = 0;
+    }
+    //checksum calculation
+    computed_checksum = computed_checksum % 256;
+    raw_data[QUANTITY_SET_VALUE - 1] = (256 - computed_checksum);
+
+    //debug
+    internal_state.packet_sent = 5;
+    data_send =5;
+    
+    expected_bytes = QUANTITY_ACK_SET_VALUES;
+    //write data
+    port->write(raw_data, QUANTITY_SET_VALUE);
+
+    return true;
+}
+
+
+/**
+ * @brief cyclically sends different Status requests to Hirth ECU
+ * 
+ * @return true - when successful
+ * @return false  - not implemented
+ */
+bool AP_EFI_Serial_Hirth::send_request_status() {
+
+    bool status = false;
+
+    switch (req_data.code)
+    {
+    case CODE_REQUEST_STATUS_1:
+        req_data.quantity = QUANTITY_REQUEST_STATUS;
+        req_data.code = CODE_REQUEST_STATUS_2;
+        req_data.checksum = CHECKSUM_REQUEST_STATUS_2;
+        expected_bytes = QUANTITY_RESPONSE_STATUS_2;
+        internal_state.packet_sent = 2;
+        data_send =2;
+        break;
+    case CODE_REQUEST_STATUS_2:
+        req_data.quantity = QUANTITY_REQUEST_STATUS;
+        req_data.code = CODE_REQUEST_STATUS_3;
+        req_data.checksum = CHECKSUM_REQUEST_STATUS_3;
+        expected_bytes = QUANTITY_RESPONSE_STATUS_3;
+        internal_state.packet_sent = 3;
+        data_send =3;
+        break;
+    case CODE_REQUEST_STATUS_3:
+        req_data.quantity = QUANTITY_REQUEST_STATUS;
+        req_data.code = CODE_REQUEST_STATUS_1;
+        req_data.checksum = CHECKSUM_REQUEST_STATUS_1;
+        expected_bytes = QUANTITY_RESPONSE_STATUS_1;
+        internal_state.packet_sent = 1;
+        data_send =1;
+        break;
+    default:
+        req_data.quantity = QUANTITY_REQUEST_STATUS;
+        req_data.code = CODE_REQUEST_STATUS_1;
+        req_data.checksum = CHECKSUM_REQUEST_STATUS_1;
+        expected_bytes = QUANTITY_RESPONSE_STATUS_1;
+        internal_state.packet_sent = 1;
+        data_send =1;
+        break;
+    }
+    //memcpy(&raw_data, 0, sizeof(raw_data));
```

Comment:
```suggestion
    //memcpy(&raw_data, 0, sizeof(raw_data));
```

probably meant memset here - crash otherwise.

Reply:
modified